enhancements/sdk-kubernetes-1.19.md

```diff
@@ -0,0 +1,249 @@
+  - Unrelated bug fixes can be done before or after this enhancement is implemented.
+  - Integration of new features introduced in 1.19 will be done in follow-on PRs.
+
+## Proposal
+
+### User Stories
+
+#### Upgrade controller-runtime to support Kubernetes 1.19
+
+_GitHub: [kubernetes-sigs/controller-runtime][gh-controller-runtime]_
+
+This story consists of:
+
+- Bumping the Kubernetes dependency versions in go.mod from 1.18 to 1.19.
+- Upgrading any relevant documentation to note support for and any breaking changes introduced by
+  the bump to Kubernetes 1.19.
+- Making any necessary changes to get CI to pass.
+
+#### Upgrade controller-tools to support Kubernetes 1.19
+
+_GitHub: [kubernetes-sigs/controller-tools][gh-controller-tools]_
+
+This story consists of:
+
+- Bumping the Kubernetes dependency versions in go.mod from 1.18 to 1.19.
+- Changing the default generated CRD version in controller-gen from v1beta1 to v1.
+- Upgrading any relevant documentation to note support for and any breaking changes introduced by
+  the bump to Kubernetes 1.19.
+- Making any necessary changes to get CI to pass.
+
+#### Upgrade kubebuilder to support Kubernetes 1.19
+
+_GitHub: [kubernetes-sigs/kubebuilder][gh-kubebuilder]_
+
+This story consists of:
+
+- Bumping the controller-runtime dependency version in go.mod file scaffolds to the version of
+  controller-runtime that supports Kubernetes 1.19.
+- Bumping the controller-gen dependency version in the Makefile scaffolds to the version of
+  controller-gen that supports Kubernete 1.19.
+- Upgrading any relevant documentation to note support for and any breaking changes introduced by
+  the bump to Kubernetes 1.19.
+- Making any necessary changes to get CI to pass.
+
+#### Upgrade helm to support Kubernetes 1.19
+
+_GitHub: [helm/helm][gh-helm]_
+
+This story consists of:
+
+- Bumping the Kubernetes dependency versions in go.mod from 1.18 to 1.19.
+- Making any necessary changes to get CI to pass.
+
+#### Upgrade prometheus-operator to support Kubernetes 1.19
+
+_GitHub: [coreos/prometheus-operator][gh-prometheus-operator]_
+
+This story consists of:
+
+- Bumping the Kubernetes dependency versions in go.mod from 1.18 to 1.19.
+- Making any necessary changes to get CI to pass.
+
+#### Upgrade operator-sdk to support Kubernetes 1.19
+
+_GitHub: [operator-framework/operator-sdk][gh-operator-sdk]_
+
+This story consists of:
+
+- Bumping the controller-runtime, controller-tools, helm, prometheus-operator, OLM, and Kubernetes
+  dependency versions in go.mod to versions that support Kubernetes 1.19.
+- Bumping the controller-runtime version in go.mod scaffolds to the controller-runtime version that
+  supports Kubernetes 1.19.
+- Updating e2e tests to use Kubernetes 1.19 where possible and create follow-on tasks for tests that
+  cannot be updated (this often happens when we update before kind images we use have been updated)
+- Making any necessary changes to get CI to pass.
+
```

Comment:
We also need to search around for SDK code that assumes a CRD manifest being unmarshaled by a non-GVK-aware unmarshaler (ex. `yaml.Unmarshal(data, &crdv1beta1)`) has version v1beta1, since this can cause some `kubectl apply` issues. I suspect that's what's happening [here](https://github.com/operator-framework/operator-sdk/issues/3005).